roles/openshift_master/tasks/main.yml

```diff
@@ -23,8 +23,9 @@
   lineinfile:
     dest: /etc/sysconfig/openshift-master
     regexp: '^OPTIONS='
-    line: "OPTIONS=\"--public-master={{ openshift_public_ip }} --nodes={{ openshift_node_ips
-          | join(',') }}  --loglevel={{ openshift_master_debug_level }}\""
+    line: "OPTIONS=\"--public-master={{ openshift_public_ip if
+          openshift_hostname_workaround else ansible_fqdn  }} --nodes={{ openshift_node_ips
+              | join(',') }}  --loglevel={{ openshift_master_debug_level }}\""
```

Comment:
@detiber Oh, or you could do that exact same filter expression as a default variable in the role. That might be nicer than doing it as set_fact. Then you can use that variable in multiple places _and_ we can use 'debug: var=newvar' to see it anywhere in the role when we're debugging. I prefer this to set_fact suggested (by me) above.